py/objtype.c

```diff
@@ -571,6 +571,10 @@ retry:;
     }
     #endif
 
+    if (op == MP_BINARY_OP_CONTAINS) {
+        return mp_obj_new_bool(mp_obj_is_true(res));
```

Comment:
I'm not 100% sure, but I think it's possible for `res` to be `MP_OBJ_NULL` at this point under certain conditions.

I think it's more correct to have this block of code up above in the case where it calls user code, just after the `res = mp_call_method_n_kw(1, 0, dest);` line.  Because user code is the only way this result could be non-boolean (C based implementations of `__contains__` should adhere to the requirement to return a boolean value).

Reply:
moved to the according place.

Can I add a test which can return `MP_OBJ_NULL` ?

Reply:
> Can I add a test which can return `MP_OBJ_NULL` ?

It's not needed anymore, but if you want to try and make such a test it would need to be something like subclassing a native type that implements `.binary_op` (at the C level), and then applying a binary op to the subclassed type that is not implemented by the native type.  Eg:
```python
class A(list):
    pass
A() ** 2
```